packages/gatsby-transformer-remark/src/on-create-node.js

```diff
@@ -0,0 +1,78 @@
+const grayMatter = require(`gray-matter`)
+const { mapValues, isDate } = require(`lodash`)
+
+const generateGraphQLType = require(`./utils/generate-graphql-type`)
+
+const mediaTypes = [`text/markdown`, `text/x-markdown`]
+
+module.exports = async function onCreateNode(helpers, pluginOptions) {
+  const {
+    node,
+    actions,
+    reporter,
+    createNodeId,
+    loadNodeContent,
+    createContentDigest,
+  } = helpers
+
+  if (mediaTypes.includes(node.internal.mediaType)) {
+    const nodeContent = await loadNodeContent(node)
+
+    try {
+      let {
+        content,
+        excerpt,
+        data: frontmatter,
+      } = grayMatter(nodeContent, pluginOptions)
+
+      if (frontmatter) {
+        frontmatter = mapValues(frontmatter, value => {
+          if (isDate(value)) {
+            return value.toJSON()
+          }
+          return value
+        })
+
+        frontmatter.title = frontmatter.title || ``
```

Comment:
@veryspry any clue why they always add a title to the frontmatter? I don't see any reason for that to be the case, especially with these new types, but I also see that it could break existing functionality which we want to avoid. Thoughts?

Reply:
I'm not 100% sure how folks are using this. Is it causing any troubles for it to exist? If not, I'd vote we leave it to stay on the safe side since the code / data is fairly minimal

Reply:
It (kind of) goes against the reason for the plugin to exist, which is to have a direct correlation between the types and the markdown files they are generated from, but I do agree that it isn't a big issue.

Having said that, I don't really see any value in it. If there is a single title in any of your markdown files, the field would exist anyways. The only difference is that in places where it isn't defined in the front matter, the field would be null rather than an empty string (which I think would be ideal).